src/mlpack/methods/ann/dists/normal_distribution.hpp

```diff
@@ -0,0 +1,155 @@
+ */
+#ifndef MLPACK_METHODS_ANN_DISTRIBUTIONS_NORMAL_DISTRIBUTION_HPP
+#define MLPACK_METHODS_ANN_DISTRIBUTIONS_NORMAL_DISTRIBUTION_HPP
+
+#include <mlpack/prereqs.hpp>
+#include "../activation_functions/logistic_function.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * Implementation of the Normal Distribution function.
+ *
+ * Normal distribution is a function which accepts a mean and a standard deviation
+ * term and creates a probablity distribution out of it.
+ */
+template <typename DataType = arma::mat>
+class NormalDistribution
+{
+ public:
+  /**
+   * Default constructor, which creates a Normal distribution with zero
+   * dimension.
+   */
+  NormalDistribution();
+
+  /**
+   * Create a Normal distribution with the given mean and sigma.
+   *
+   * @param mean The mean of the normal distribution.
+   * @param sigma The standard deviation of the normal distribution.
+   */
+  NormalDistribution(const DataType& mean, const DataType& sigma);
+
+  /**
+   * Return the probabilities of the given matrix of observations.
+   *
+   * @param observation The observation matrix.
+   */
+  DataType Probability(const DataType& observation) const
+  {
+    return arma::exp(LogProbability(observation));
+  }
+
+  /**
+   * Return the log probabilities of the given matrix of observations.
+   *
+   * @param observation The observation matrix.
+   */
+  DataType LogProbability(const DataType& observation) const;
+
+  /**
+   * Stores the gradient of the probabilities of the observations
+   * with respect to mean and standard deviation.
+   *
+   * @param observation The observation matrix.
+   * @param dmu The gradient with respect to mean.
+   * @param dsigma The gradient with respect to standard deviation.
+   */
+  void ProbBackward(const DataType& observation,
+                    DataType& dmu,
+                    DataType& dsigma) const;
+
+  /**
+   * Calculates the normal probability density function for each
+   * data point (column) in the given matrix.
+   *
+   * @param x List of observations.
+   * @param probabilities Output probabilities for each input observation.
+   */
+  void Probability(const DataType& x, DataType& probabilities) const
+  {
+    probabilities = Probability(x);
+  }
+
+  /**
+    * Calculates the log of normal probability density function for each
+    * data point (column) in the given matrix.
+    *
+    * @param x List of observations.
+    * @param log probabilities Output probabilities for each input observation.
+    */
+  void LogProbability(const DataType& x, DataType& probabilities) const
+  {
+    probabilities = LogProbability(x);
+  }
+
+  /**
+   * Return a randomly generated observation according to the probability
+   * distribution defined by this object.
+   *
+   * @return Random observation from this Normal distribution.
+   */
+  DataType Sample() const;
+
+  /**
+   * Return the mean.
+   */
```

Comment:
Hey @nishantkr18, For single line comments, do you mind using `//` to comment. Thanks a lot!